app/helpers/email_helper.rb

```diff
@@ -29,11 +29,70 @@ def email_html_template
     EmailStyle.new.html
       .sub('%{email_content}') { capture { yield } }
       .gsub('%{html_lang}', html_lang)
+      .gsub('%{dark_mode_meta_tags}', SiteSetting.dark_mode_emails_active ? dark_mode_meta_tags : "")
+      .gsub('%{dark_mode_styles}', SiteSetting.dark_mode_emails_active ? dark_mode_styles : "")
       .html_safe
   end
 
   protected
 
+  def dark_mode_meta_tags
+    "
+    <meta name='color-scheme' content='light dark' />
+    <meta name='supported-color-schemes' content='light dark' />
+    "
+  end
+
+  def dark_mode_styles
+    "
+    <style>
+      @media (prefers-color-scheme: dark) {
+        html {
+          background: #151515 !important;
+        }
+
+        h1,
+        h2,
+        h3,
+        h4,
+        h5,
+        h6,
+        p,
+        span {
+          color: #dddddd !important;
+        }
+
+        [dm='light-img'] {
+          display: none !important;
+        }
+
+        [dm='dark-img'] {
+          display: block !important;
+          opacity: 0.7 !important;
```

Comment:
We probably shouldn't tweak opacity here.

Reply:
We need to override the default opacity that is being applied, otherwise we will end up with this: 

<img width="977" alt="Screen Shot 2022-04-05 at 12 21 21 PM" src="https://user-images.githubusercontent.com/50783505/161814167-e6ebc3f7-ff8d-462a-a14d-706a4caccb7c.png">

instead of: 

<img width="942" alt="Screen Shot 2022-04-05 at 12 22 18 PM" src="https://user-images.githubusercontent.com/50783505/161814242-7c608ed6-4f6a-4f5d-b89b-6b38fc0de4e6.png">